src/Analyzers/CSharp/Tests/UseDeconstruction/UseDeconstructionTests.cs

```diff
@@ -3,818 +3,914 @@
+            var code = """
+                class C
+                {
+                    void M()
+                    {
+                        var t1 = GetPerson();
+                        {
+                            int age;
+                        }
+                    }
+
+                    (string name, int age) GetPerson() => default;
+                }
+                """;
 
-    (string name, int age) GetPerson() => default;
-}");
+            await VerifyCS.VerifyCodeFixAsync(code, code);
         }
 
         [Fact]
         public async Task TestNotIfNameInOuterScope()
         {
-            await TestMissingInRegularAndScriptAsync(
-@"class C
-{
-    int age;
+            var code = """
+                class C
+                {
+                    int age;
 
-    void M()
-    {
-        var [|t1|] = GetPerson();
-    }
+                    void M()
+                    {
+                        var t1 = GetPerson();
+                    }
+
+                    (string name, int age) GetPerson() => default;
+                }
+                """;
 
-    (string name, int age) GetPerson() => default;
-}");
+            await VerifyCS.VerifyCodeFixAsync(code, code);
         }
 
         [Fact]
         public async Task TestUpdateReference()
         {
-            await TestInRegularAndScript1Async(
-@"class C
-{
-    void M()
-    {
-        var [|t1|] = GetPerson();
-        Console.WriteLine(t1.name + "" "" + t1.age);
-    }
+            await VerifyCS.VerifyCodeFixAsync("""
+                class C
+                {
+                    void M()
+                    {
+                        var [|t1|] = GetPerson();
+                        System.Console.WriteLine(t1.name + " " + t1.age);
+                    }
 
-    (string name, int age) GetPerson() => default;
-}",
-@"class C
-{
-    void M()
-    {
-        var (name, age) = GetPerson();
-        Console.WriteLine(name + "" "" + age);
-    }
+                    (string name, int age) GetPerson() => default;
+                }
+                """, """
+                class C
+                {
+                    void M()
+                    {
+                        var (name, age) = GetPerson();
+                        System.Console.WriteLine(name + " " + age);
+                    }
 
-    (string name, int age) GetPerson() => default;
-}");
+                    (string name, int age) GetPerson() => default;
+                }
+                """);
         }
 
         [Fact]
         public async Task TestTupleType()
         {
-            await TestInRegularAndScript1Async(
-@"class C
-{
-    void M()
-    {
-        (int name, int age) [|t1|] = GetPerson();
-        Console.WriteLine(t1.name + "" "" + t1.age);
-    }
+            await VerifyCS.VerifyCodeFixAsync("""
+                class C
+                {
+                    void M()
+                    {
+                        (string name, int age) [|t1|] = GetPerson();
```

Comment:
There is a `TestTupleTypeInForEach`, that verifies the same thing for `foreach` and it doesn't have tuple type mismatch. So I consider type mismatch here as unintended noise error